src/components/Icon/Icon.tsx

```diff
@@ -5,68 +5,149 @@ import svg4everybody from 'svg4everybody';
 import styles from './Icon.module.css';
 import icons from '../../icons/spritemap/spritemap.svg';
 
-export interface Props {
+interface IconPropsBase {
   /**
    * CSS class names that can be appended to the component.
    */
   className?: string;
   /**
-   * Declares whether the icon is focusable or not for accessibility
+   * Either a fragment of svg elements (g, path, circle, etc.), or a single
+   * svg element. Useful for creating specific icon components.
+   *
+   * @example
+   * function CircleIcon(props: IconProps) {
+   *   return (
+   *     <Icon {...props}>
+   *       <circle cx="50" cy="50" r="50" />
+   *     </Icon>
+   *   )
+   * }
    */
-  focusable?: boolean;
+  children?: React.ReactNode;
+  /**
+   * The SVG Color, expects a valid css color (hex, rgb, etc.).
+   * Recommendation: if `currentColor` isn't sufficient, use color tokens from
+   * @chanzuckerberg/eds/lib/tokens/colors.ts
+   */
+  color?: string;
+  /**
+   * If true, the element will be displayed as a block, otherwise
+   * default is inline like images
+   */
+  fullWidth?: boolean;
   /**
    * ID used so the svg can read the title of the SVG icon to the user when accessibility is needed
    */
-  id?: any;
+  id?: string;
+  /**
+   * Inverted variant for dark backgrounds
+   *
+   * TODO: investigate if needed
+   */
+  inverted?: boolean;
   /**
    * Name of icon to reference in icon sprite
+   *
+   * TODO: add typing of possible icon names
    */
   name?: string;
   /**
-   * Inverted variant for dark backgrounds
+   * Width/Height string (px, rem, em, vh, etc.)
+   * Recommendation: use "EdsFontSize" tokens from
+   * @chanzuckerberg/eds/lib/tokens/variables.json
    */
-  inverted?: boolean;
+  size?: string;
+}
+
+interface InformativeIconProps extends IconPropsBase {
   /**
-   * SVG title which serves as alt text for the SVG.
+   * The role of the icon.
+   *
+   * Use "informative" when the icon **_does_** provide additional meaning to other text on the
+   * page. You'll be required to pass in a title to label the icon.
    */
-  title?: string;
+  purpose: 'informative';
+  title: string;
+}
+
+interface DecorativeIconProps extends IconPropsBase {
+  /**
+   * The role of the icon.
+   *
+   * Use "decorative" when the icon **_does not_** provide any additional context or meaning to
+   * associated text. Basically the icon is for show and people don't need it to understand what's
+   * on the page.
+   */
+  purpose: 'decorative';
+}
+
+export type IconProps = DecorativeIconProps | InformativeIconProps;
```

Comment:
I thiiink unions have to be declared as `type`, otherwise this would need to look something like `interface IconProps extends Decorative..., Informative... {}`

but I don't really know the differences between type and interface in the first place 😅